src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -1961,18 +1946,23 @@ JRT_ENTRY(void, Deoptimization::uncommon_trap_inner(JavaThread* current, jint tr
         xtty->end_head();
       }
       if (TraceDeoptimization) {  // make noise on the tty
-        tty->print("Uncommon trap occurred in");
-        nm->method()->print_short_name(tty);
-        tty->print(" compiler=%s compile_id=%d", nm->compiler_name(), nm->compile_id());
+        stringStream st;
+        st.print("UNCOMMON TRAP method=%s", trap_scope->method()->name_and_sig_as_C_string());
+        st.print("  bci=%d pc=" INTPTR_FORMAT ", relative_pc=" INTPTR_FORMAT JVMCI_ONLY(", debug_id=%d"), trap_scope->bci(), p2i(fr.pc()), fr.pc() - nm->code_begin()
+#if INCLUDE_JVMCI
+            , debug_id
+#endif
```

Comment:
You can use `JVMCI_ONLY` here as well. Same in line 1840.

Reply:
Unfortunately, because of the comma and the way JVMCI_ONLY is defined, this does not work. I will leave it as it is